src/alpaka/test/alpaka/AtomicPairCounter_t.cc

```diff
@@ -0,0 +1,138 @@
+#include <cassert>
+#include <iostream>
+
+#include "AlpakaCore/alpakaConfig.h"
+#include "AlpakaCore/AtomicPairCounter.h"
+
+
+using namespace ALPAKA_ACCELERATOR_NAMESPACE;
+
+struct update {
+  template <typename T_Acc>
+  ALPAKA_FN_ACC void operator()(const T_Acc& acc, cms::Alpaka::AtomicPairCounter *dc, uint32_t *ind, uint32_t *cont, uint32_t n) const {
+    // Global thread index in grid
+    const uint32_t threadIdxGlobal(alpaka::idx::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u]);
+    const uint32_t threadDimension(alpaka::workdiv::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[0u]);
+
+    // Global element index (obviously relevant for CPU only, for GPU, i = threadIndexGlobal only)
+    const uint32_t firstElementIdxGlobal = threadIdxGlobal * threadDimension;
+    const uint32_t endElementIdxGlobalUncut = firstElementIdxGlobal + threadDimension;
+    const uint32_t endElementIdxGlobal = std::min(endElementIdxGlobalUncut, n);
```

Comment:
(possibly beyond this PR)

I'm wondering if it would make sense already now to abstract these with something along
```cpp
template <typename T, typename T_Acc>
std::pair<T, T> make_global_index_range_max(const T_Acc& acc, T max) {
  // Global thread index in grid
  const T threadIdxGlobal(alpaka::idx::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u]);
  const T threadDimension(alpaka::workdiv::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[0u]);

  // Global element index (obviously relevant for CPU only, for GPU, i = threadIndexGlobal only)
  const T firstElementIdxGlobal = threadIdxGlobal * threadDimension;
  const T endElementIdxGlobalUncut = firstElementIdxGlobal + threadDimension;
  const T endElementIdxGlobal = std::min(endElementIdxGlobalUncut, max);

  return {firstElementIdxGlobal, endElementIdxGlobal};
}
```
and call here
```cpp
  const auto [firstElementIdxGlobal, endElementIdxGlobal] = cms::alpaka::make_global_index_range_max(acc, n);
```

(or instead of `pair` return a wrapper that provides `begin()` and `end()` so the returned object could be used with range-for)

Reply:
Yep absolutely